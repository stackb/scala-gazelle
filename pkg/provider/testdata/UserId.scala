package common.types

/** Scala "Value" class for interacting with UserId values
  * Representative of conceptual ActorId/UserId primary key Int values
  *
  * @param value In general it should not be necessary or desirable to interact directly with this property. Exceptions
  *              being Core Libraries, DB interactions, and legacy code (which should be updated opportunistically).
  */
case class UserId(value: Int) extends AnyVal

object UserId {
  implicit val intTypeMapper: scalapb.TypeMapper[Int, UserId] = scalapb.TypeMapper[Int, UserId](UserId(_))(_.value)
}
