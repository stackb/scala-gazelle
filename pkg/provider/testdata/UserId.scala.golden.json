{
  "label": "//pkg/provider:testdata/UserId.scala",
  "kind": "scala_library",
  "files": [
    {
      "filename": "testdata/UserId.scala",
      "packages": [
        "common.types"
      ],
      "classes": [
        "common.types.UserId"
      ],
      "objects": [
        "common.types.UserId"
      ],
      "names": [
        "AnyVal",
        "Int",
        "TypeMapper",
        "UserId",
        "scalapb.TypeMapper"
      ],
      "extends": {
        "class common.types.UserId": {
          "classes": [
            "AnyVal"
          ]
        }
      }
    }
  ]
}